File: app/src/main/java/fr/android/androidexercises/Step0Fragment.java
package fr.android.androidexercises;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

public class Step0Fragment extends Fragment {

    private static final String STEP_0 = "This is step 0";

    private TextView textView;
    private OnNextStep0Listener listener;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        // TODO cast context to listener
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_step0, container, false);

        // TODO find TextView and set text

        // TODO find Button and set listener
        Button nextButton;
        nextButton.setOnClickListener(v -> {
            // TODO call listener
        });

        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        // TODO setText(STEP_0)
    }

    public interface OnNextStep0Listener {
        // TODO add onNext() method
    }

}

File: app/src/main/java/fr/android/androidexercises/Step0Fragment.kt
package fr.android.androidexercises

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView

class Step0Fragment : Fragment() {

    companion object {
        private val STEP_0 = "This is step 0"
    }

    private var textView: TextView? = null
    private var listener: OnNextStep0Listener? = null

    override fun onAttach(context: Context?) {
        super.onAttach(context)

        // TODO cast context to listener
        when (context) {
            is OnNextStep0Listener -> listener = context
            else -> throw Exception("erreur")
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_step0, container, false)

        // TODO find TextView and set text
        textView = view.findViewById(R.id.step1TextView)

        // TODO find Button and set listener
        val nextButton: Button = view.findViewById(R.id.nextButton)
        nextButton.setOnClickListener {
            // TODO call listener
            listener?.onNext()
        }

        return view
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        // TODO setText(STEP_0)
        textView?.setText(STEP_0)
    }

    interface OnNextStep0Listener //TODO add onNext() method
    {
        fun onNext()
    }

}
